lib/solidus_dev_support/rspec/feature_helper.rb

```diff
@@ -22,18 +22,27 @@
 
 require 'spree/testing_support/capybara_ext'
 
+def dev_support_assets_preload
+  if Rails.application.respond_to?(:precompiled_assets)
+    Rails.application.precompiled_assets
+  else
+    # For older sprockets 2.x
+    Rails.application.config.assets.precompile.each do |asset|
+      Rails.application.assets.find_asset(asset)
+    end
+  end
+end
+
 RSpec.configure do |config|
   config.when_first_matching_example_defined(type: :feature) do
     config.before :suite do
-      # Preload assets
-      if Rails.application.respond_to?(:precompiled_assets)
-        Rails.application.precompiled_assets
-      else
-        # For older sprockets 2.x
-        Rails.application.config.assets.precompile.each do |asset|
-          Rails.application.assets.find_asset(asset)
-        end
-      end
+      dev_support_assets_preload
+    end
+  end
+
+  config.when_first_matching_example_defined(type: :system) do
+    config.before :suite do
+      dev_support_assets_preload
```

Comment:
What about using https://github.com/solidusio/solidus/blob/138bf619647b125a53c8898fce6d01a9dee2c7a2/core/lib/spree/testing_support/precompiled_assets.rb for Solidus >= 2.11?